bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,219 @@
+                                                    yioremoteDockwebSocketClient
+                                                            .sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+
+                                                    Thread.sleep(1000);
+
+                                                    if (yioremoteDockwebSocketClient.getbooleanheartbeat()) {
+                                                        logger.debug("heartbeat ok");
+                                                        triggerChannel(getChannelUuid(GROUP_OUTPUT, YIODOCKSTATUS));
+                                                        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS,
+                                                                yioremoteDockwebSocketClient.getstringreceivedstatus());
+                                                    } else {
+                                                        yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTIONFAILED;
+                                                        updateStatus(ThingStatus.OFFLINE,
+                                                                ThingStatusDetail.CONFIGURATION_ERROR,
+                                                                "Connection lost no ping from YIO DOCK");
+                                                        updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                                                        pollingJob.cancel(true);
+                                                    }
+                                                } else {
+                                                    yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTIONFAILED;
+                                                    updateStatus(ThingStatus.OFFLINE,
+                                                            ThingStatusDetail.CONFIGURATION_ERROR,
+                                                            "Connection lost no ping from YIO DOCK");
+                                                    updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                                                    pollingJob.cancel(true);
+                                                }
+                                            } catch (Exception e) {
+                                                logger.warn("Error during initializing the WebSocket polling Thread {}",
+                                                        e.getMessage());
+                                            }
+                                        }
+                                    };
+
+                                    pollingJob = scheduler.scheduleWithFixedDelay(heartbeatpolling, 0, 30,
+                                            TimeUnit.SECONDS);
+                                } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
Changed complete code